fieldtracing/fieldtracing.h

```diff
@@ -0,0 +1,207 @@
+      Real fte_max_curvature_radii_to_trace;
+      Real fte_max_curvature_radii_extent;
+      Real fte_max_m_to_trace;
+   };
+   
+   extern FieldTracingParameters fieldTracingParameters;
+   
+   /*! Type of field line ending, used to classify the ionospheric nodes and the forward and backward field lines in full.box tracing.
+   * CLOSED: ends in the ionosphere
+   * OPEN: exits the simulation domain
+   * LOOP: has not exited, might keep looping or would exit given enough time/steps
+   * UNPROCESSED: cells inside the ionosphere or outside the outer limits that weren't even processed in the first place
+   */
+   enum TracingLineEndType {
+      CLOSED,
+      OPEN,
+      LOOP,
+      UNPROCESSED // Keep last for the reductions to work!
+   };
+
+   /*! Type of connection for a point traced forward and backward.
+   * The first is for the forward end, the second for the backward end.
+   * Used for full-box tracing.
+   * See TracingLineEndTypes for explanation of types.
+   * INVALID used for UNPROCESSED or other unparsed values ==> bug?
+   * \sa TracingLineEndTypes
+   */
+   enum TracingPointConnectionType {
+      CLOSED_CLOSED,
+      CLOSED_OPEN,
+      OPEN_CLOSED,
+      OPEN_OPEN,
+      CLOSED_LOOP,
+      LOOP_CLOSED,
+      OPEN_LOOP,
+      LOOP_OPEN,
+      LOOP_LOOP,
+      INVALID // Keep last for the reductions to work!
+   };
+
+   /*! Simple method to translate 3D to 1D indices */
+   inline int ijk2Index(
+      int i,
+      int j,
+      int k,
+      std::array<int,3>dims
+   ) {
+      return i + j*dims[0] +k*dims[0]*dims[1];
+   }
+
+   /*! Handler function for field line tracing */
+   typedef std::function<bool(std::array<Real,3>&, const bool, std::array<Real, 3>&)> TracingFieldFunction;
+   
+   void stepFieldLine(
+      std::array<Real, 3>& x,
+      std::array<Real, 3>& v,
+      Real& stepSize,
+      creal minStepSize,
+      creal maxStepSize,
+      TracingMethod method,
+      TracingFieldFunction& BFieldFunction,
+      const bool outwards=true
+   );
+   
+   /*! function to empty the Balsara reconstruction coefficient cache at a new time step */
+   inline void resetReconstructionCoefficientsCache() {
+      fieldTracingParameters.reconstructionCoefficientsCache.clear();
+   }
+   
+   /*! Link each ionospheric node to fsgrid cells for coupling */
+   void calculateFsgridCoupling(
+      FsGrid< fsgrids::technical, FS_STENCIL_WIDTH> & technicalGrid,
+      FsGrid< std::array<Real, fsgrids::bfield::N_BFIELD>, FS_STENCIL_WIDTH> & perBGrid,
+      FsGrid< std::array<Real, fsgrids::dperb::N_DPERB>, FS_STENCIL_WIDTH> & dPerBGrid,
+      std::vector<SBC::SphericalTriGrid::Node> & nodes,
+      creal radius
+   );
+
+   /*! Find coupled ionosphere mesh node for given location */
+   std::array<std::pair<int, Real>, 3> calculateVlasovGridCoupling(
```

Comment:
Since this is now no longer in the ionosphere namespace or file, I suggest the name should be changed to "calculateVlasovIonosphereCoupling", just to make it clear this function deals with the ionosphere grid.